Sources/MongoSwift/ConnectionString.swift

```diff
@@ -5,45 +5,33 @@ internal class ConnectionString {
     /// Pointer to the underlying `mongoc_uri_t`.
     internal let _uri: OpaquePointer
 
-    /// Initializes a new `ConnectionString` with the provided options. Updates `options` to correctly reflect the
-    /// final options set on the `ConnectionString`.
-    internal init(_ connectionString: String, options: inout ClientOptions) throws {
+    /// Initializes a new `ConnectionString` with the provided options.
+    internal init(_ connectionString: String, options: ClientOptions? = nil) throws {
```

Comment:
for some reason I wrote this stupid `inout` API where the connection string would update the client options struct to reflect the final values on the URI so that the `MongoClient` could have correct values for read concern, write concern, and read preference.
however, it's way simpler to just have `MongoClient` read back those values by calling the appropriate getters on the connection string after it creates it (see `MongoClient.init`.)